sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedFluxCore.java

```diff
@@ -156,23 +152,25 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
      * @param state the state to be used across multiple Page Retriever Function calls
      * @param pageRetriever the Page Retriever Function
      * @param pageSize the preferred page size
-     * @param <C> the type of Continuation token
-     * @param <T> The type of items in a {@link ContinuablePage}
-     * @param <P> The {@link ContinuablePage} holding items of type {@code T}
      * @return a Flux of {@link ContinuablePage}
      */
-    private static <C, T, P extends ContinuablePage<C, T>> Flux<P> concatFluxOfPage(ContinuationState<C> state,
-        PageRetriever<C, P> pageRetriever, Integer pageSize) {
+    private Flux<P> retrievePages(ContinuationState<C> state, PageRetriever<C, P> pageRetriever, Integer pageSize) {
+        return retrievePage(state, pageRetriever, pageSize)
+            .expand(page -> {
+                state.setLastContinuationToken(page.getContinuationToken());
+                return Flux.defer(() -> retrievePage(state, pageRetriever, pageSize));
+            }, 4);
```

Comment:
Any reason for setting this to 4 instead of 1? Probably adding a comment to explain why 4 was chosen would help.

Reply:
I should add a comment here. The second argument is a hint to the operator that determines how large of an initial buffer it should instantiate. Given that `PageRetriever` returns a `Flux` the implementation could return more than one.